### ISSUE TITLE
SMTP settings form edit bug

### ISSUE THREAD

**Issue by remydavid** [metabase/metabase#4178]
- Your browser and the version: Chrome 55
- Your operating system: Windows 7
- Your databases: BigQuery
- Metabase version: 21.1
- Metabase hosting environment: App Engine
- Metabase internal database: Google SQL

I tried to add my Gmail account for SMTP. First it worked, but when I clicked test mail I got an error, so I came back to the form to edit the settings, now when I save I got the following error : 

`Duplicate entry 'email-smtp-host' for key 'PRIMARY'`

Logs : 

```clojure
Jan 17 09:09:33 ERROR metabase.middleware :: PUT /api/email 500 (606 ms) (0 DB calls)
{:message "Duplicate entry 'email-smtp-host' for key 'PRIMARY'",
 :stacktrace ["db$simple_insert_many_BANG_.invokeStatic(db.clj:725)"
              "db$simple_insert_many_BANG_.invoke(db.clj:712)"
              "db$simple_insert_BANG_.invokeStatic(db.clj:750)"
              "db$simple_insert_BANG_.invoke(db.clj:740)"
              "db$insert_BANG_.invokeStatic(db.clj:765)"
              "db$insert_BANG_.invoke(db.clj:754)"
              "db$insert_BANG_.invokeStatic(db.clj:768)"
              "db$insert_BANG_.doInvoke(db.clj:754)"
              "models.setting$fn__17189$set_string_BANG___17194$fn__17195.invoke(setting.clj:188)"
              "models.setting$fn__17189$set_string_BANG___17194.invoke(setting.clj:173)"
              "models.setting$set_BANG_.invokeStatic(setting.clj:227)"
              "models.setting$set_BANG_.invoke(setting.clj:218)"
              "models.setting$set_many_BANG_.invokeStatic(setting.clj:330)"
              "models.setting$set_many_BANG_.invoke(setting.clj:323)"
              "api.email$fn__23747$fn__23749.invoke(email.clj:69)"
              "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:222)"
              "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:219)"
              "api.email$fn__23747.invokeStatic(email.clj:54)"
              "api.email$fn__23747.invoke(email.clj:54)"
              "middleware$enforce_authentication$fn__31028.invoke(middleware.clj:84)"
              "api.routes$fn__31136.invokeStatic(routes.clj:44)"
              "api.routes$fn__31136.invoke(routes.clj:44)"
              "routes$fn__32124.invokeStatic(routes.clj:24)"
              "routes$fn__32124.invoke(routes.clj:24)"
              "middleware$log_api_call$fn__31116$fn__31118.invoke(middleware.clj:275)"
              "db$_do_with_call_counting.invokeStatic(db.clj:461)"
              "db$_do_with_call_counting.invoke(db.clj:455)"
              "middleware$log_api_call$fn__31116.invoke(middleware.clj:274)"
              "middleware$add_security_headers$fn__31075.invoke(middleware.clj:209)"
              "middleware$bind_current_user$fn__31032.invoke(middleware.clj:104)"],
 :sql-exception-chain ["MySQLIntegrityConstraintViolationException:" "Message: Duplicate entry 'email-smtp-host' for key 'PRIMARY'" "SQLState: 23000" "Error Code: 1062"]}
```

**Comment by camsaul** [comment]
As noted in #4673 this issue can also affect other settings, for example Slack

**Comment by camsaul** [comment]
Fixed by #4674, going out in our next release
